backend/tests/Feature/ModuleHookEventsTest.php

```
@@ -0,0 +1,127 @@
+        $step->update(['status' => BatchStep::STATUS_IN_PROGRESS]);
+        Event::assertDispatched(StepStarted::class, fn ($e) => $e->batchStep->is($step));
+
+        $step->update(['status' => BatchStep::STATUS_DONE]);
+        Event::assertDispatched(StepCompleted::class, fn ($e) => $e->batchStep->is($step));
+    }
+
+    public function test_resource_changed_fires_for_curated_entities(): void
+    {
+        Event::fake([ResourceChanged::class]);
+
+        $customer = Customer::factory()->create();
```

PHPStan (2.2.6)

[error] Call to an undefined static method App\Models\Customer::factory().


(staticMethod.notFound)